{{cookiecutter.repo_name}}/{{cookiecutter.package_name}}/utils/models.py

```diff
@@ -0,0 +1,69 @@
+from __future__ import print_function
+
+from django.template.loader import render_to_string
+from django.urls import reverse
+from django.utils.functional import cached_property
+from django.utils.safestring import mark_safe
+from django_lazy_image.templatetags.lazy_image import lazy_image
+from threadlocals.threadlocals import get_current_user
+
+
+class EditableMixin:
+    def get_displayed_text_for_CharField(self, field_name, field_value):
+        # Need to handle max_length
+        return mark_safe(f'<span class="js-SimpleEditable" data-pk="{self.pk}" data-field="{field_name}" data-app="{self._meta.app_label}" data-model="{self._meta.object_name}" contentEditable>{field_value}</span>')
+
+    def get_displayed_text_for_TextField(self, field_name, field_value):
+        return mark_safe(f'<span class="js-SimpleEditable" data-pk="{self.pk}" data-field="{field_name}" data-app="{self._meta.app_label}" data-model="{self._meta.object_name}" contentEditable>{field_value}</span>')
+
+    def get_displayed_text_for_AutoField(self, field_name, field_value):
```

Comment:
Would you ever edit an AutoField?

Reply:
Dunno, I don't think it hurts to have the option?

Reply:
An AutoField is an auto-incrementing ID enforced by the database, as used in the `id` column on models and just about nowhere else - it is designed to be a permanent, immutable identifier. Even if it was ever rendered on the front-end I think it would be dangerous to allow it to be edited.

Reply:
An example of when editing this would be actively harmful, by the way. Say the highest number in an AutoField was 18, setting the next one to 19. If you *did* edit 18 to change it to 20, the database would probably happily let you do that. But as I recall (from having to fix this in the past from data imports gone bad) *this would not affect the column's sequence counter*. It would stay at 19, then get incremented to 20 on the next insertion, and on the *next* insertion you would encounter a uniqueness constraint violation as the database still thinks the next in the sequence is 20.